packages/block-library/src/image/view.js

```diff
@@ -137,7 +135,42 @@ store(
 							false
 						);
 					},
-					hideLightbox: async ( { context } ) => {
+					// When opening the lightbox via clicking an image,
+					// we can use the event target directly and pass it to the
+					// showLightbox action, which uses it to create the styles.
+					callShowLightboxFromImage: ( {
+						context,
+						event,
+						actions,
+					} ) => {
+						actions.core.image.showLightbox(
+							{
+								context,
+								event,
+							},
+							event.target
+						);
+					},
+					// When opening the lightbox via clicking the button,
+					// we need to reach into event target's parent element to
+					// get the image element needed to create the styles.
+					callShowLightboxFromButton: ( {
+						context,
+						event,
+						actions,
+					} ) => {
+						actions.core.image.showLightbox(
+							{
+								context,
+								event,
+							},
+							// The event target we receive when clicking the button
+							// is the SVG element inside of it, so we need to go to
+							// the parent element's sibling to get the image.
+							event.target.parentElement.previousElementSibling
+						);
+					},
```

Comment:
Can't we store the `<img>` element ref in the context and use it [here](https://github.com/WordPress/gutenberg/blob/cea71f1e57712d370e503e5232a806e6fda562ed/packages/block-library/src/image/view.js#L109)? Something like:

```js
setStyles( context, context.core.image.imageRef );
```

It feels like we are creating different actions serving the same purpose.

Reply:
Yup, you're right! I've taken your suggestion and cleaned it up.